src/Workspaces/Core/MSBuild/MSBuild/DiagnosticReporter.cs

```diff
@@ -58,13 +58,24 @@ public void Report(IEnumerable<DiagnosticLogItem> log)
         {
             foreach (var logItem in log)
             {
-                Report(DiagnosticReportingMode.Log, GetMSBuildFailedMessage(logItem.ProjectFilePath, logItem.ToString()));
+                var diagnosticKind = logItem.Kind is DiagnosticLogItemKind.Warning ? WorkspaceDiagnosticKind.Warning : WorkspaceDiagnosticKind.Failure;
+
+                Report(new WorkspaceDiagnostic(diagnosticKind, GetMSBuildFailedMessage(logItem.ProjectFilePath, logItem.ToString(), diagnosticKind)));
             }
         }
 
-        private static string GetMSBuildFailedMessage(string projectFilePath, string message)
+        private static string GetMSBuildFailedMessage(string projectFilePath, string message, WorkspaceDiagnosticKind diagnosticKind)
             => RoslynString.IsNullOrWhiteSpace(message)
```

Comment:
It might just be easier to fill in a blank message with "no message" or something so the log entry would be a bit clearer we're missing one (and it reduces the number of strings we have). That said, happy to keep this as is since you're just maintaining behavior.